src/Microsoft.AspNetCore.Mvc.RazorPages/ApplicationModels/PageHandlerModel.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.AspNetCore.Mvc.ApplicationModels
+{
+    /// <summary>
+    /// Represents a handler in a <see cref="PageApplicationModel"/>.
+    /// </summary>
+    [DebuggerDisplay("PageHandlerModel: Name={" + nameof(PageHandlerModel.Name) + "}")]
+    public class PageHandlerModel : ICommonModel
+    {
+        /// <summary>
+        /// Creates a new <see cref="PageHandlerModel"/>.
+        /// </summary>
+        /// <param name="handlerMethod">The <see cref="System.Reflection.MethodInfo"/> for the handler.</param>
+        /// <param name="attributes">Any attributes annotated on the handler method.</param>
+        public PageHandlerModel(
+            MethodInfo handlerMethod,
+            IReadOnlyList<object> attributes)
+        {
+            MethodInfo = handlerMethod ?? throw new ArgumentNullException(nameof(handlerMethod));
+            Attributes = attributes ?? throw new ArgumentNullException(nameof(attributes));
```

Comment:
Don't worry, one of these days all our null checks are going to look like this 😈